rcl/test/test_namespace.cpp

```diff
@@ -33,17 +33,28 @@ using namespace std::chrono_literals;
 class TestNamespaceFixture : public ::testing::Test
 {
 public:
+  rcl_context_t * context_ptr;
   rcl_node_t * node_ptr;
   void SetUp()
   {
     rcl_ret_t ret;
-    ret = rcl_init(0, nullptr, rcl_get_default_allocator());
-    ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    {
+      rcl_init_options_t init_options = rcl_get_zero_initialized_init_options();
+      ret = rcl_init_options_init(&init_options, rcl_get_default_allocator());
+      ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+      OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT({
+        EXPECT_EQ(RCL_RET_OK, rcl_init_options_fini(&init_options)) << rcl_get_error_string().str;
+      });
+      this->context_ptr = new rcl_context_t;
```

Comment:
Is there a particular reason to use `new`? We could also declare as a non-pointer type and pass a reference below.

The same for the node handle as well.

Reply:
Mostly because that's how it was before (node was a pointer setup with new/delete). I think there might have been a reason for it, but I don't remember now.